x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/dashboards_buttons.tsx

```diff
@@ -5,13 +5,37 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useEffect } from 'react';
 import { EuiFlexGroup, EuiFlexItem, EuiButtonEmpty } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n-react';
 
 import { DASHBOARD_LOCATORS_IDS } from '../../../../../../../common/constants';
 
-import { useDashboardLocator } from '../../../../hooks';
+import { useDashboardLocator, useStartServices } from '../../../../hooks';
+
+const useDashboardExists = (dashboardId: string) => {
```

Comment:
as this hook is only used here I kept it in the same file, happy to move it to its own file?

Reply:
I think it can stay in this file and can be moved if we need to reuse it elsewhere